### ISSUE TITLE
Android library not finding native method nSetBitmap

### ISSUE THREAD

**Issue by rawnsley** [google/filament#4905]
I've updated my fork from 1.12.10 to 1.14.2 and I get a runtime error when I try and use `TextureHelper.setBitmap`:

```
E/art: No implementation found for void com.google.android.filament.android.TextureHelper.nSetBitmap(long, long, int, int, int, int, int, android.graphics.Bitmap, int) (tried Java_com_google_android_filament_android_TextureHelper_nSetBitmap and Java_com_google_android_filament_android_TextureHelper_nSetBitmap__JJIIIIILandroid_graphics_Bitmap_2I)
E/HUB: [UncaughtExceptionHandler]
    Exception[0]: No implementation found for void com.google.android.filament.android.TextureHelper.nSetBitmap(long, long, int, int, int, int, int, android.graphics.Bitmap, int) (tried Java_com_google_android_filament_android_TextureHelper_nSetBitmap and Java_com_google_android_filament_android_TextureHelper_nSetBitmap__JJIIIIILandroid_graphics_Bitmap_2I)
W/System.err: java.lang.UnsatisfiedLinkError: No implementation found for void com.google.android.filament.android.TextureHelper.nSetBitmap(long, long, int, int, int, int, int, android.graphics.Bitmap, int) (tried Java_com_google_android_filament_android_TextureHelper_nSetBitmap and Java_com_google_android_filament_android_TextureHelper_nSetBitmap__JJIIIIILandroid_graphics_Bitmap_2I)
        at com.google.android.filament.android.TextureHelper.nSetBitmap(Native Method)
        at com.google.android.filament.android.TextureHelper.setBitmap(TextureHelper.java:65)
        at com.google.android.filament.android.TextureHelper.setBitmap(TextureHelper.java:40)
        at avn.graphics.fframe.materials.MaterialLibrary$sharedPlaceholderTexture$1$1.invokeSuspend(MaterialLibrary.kt:72)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:761)
```
I can't see any obvious reason why it can't find it to link to and I've tried a clean build. Incidentally I have to disable _org.gradle.unsafe.configuration-cache_ to make the build work reliably.

It's quite possibly "user error", but I thought I would ask in case I've missed some change I should make to my build process or application initialization.

I'm doing a clean build of the main branch without my changes to be sure it's not me, but I have to unpick some fork-specific calls in my code so it takes a while.

**Comment by romainguy** [comment]
I can reproduce the issue but I don't understand it.

**Comment by romainguy** [comment]
I found the issue and it's a big wtf.

**Comment by romainguy** [comment]
tl;dr Looks like a bug in the build system (AGP and/or NDK) is causing the `-DANDROID` argument to be dropped from the default set of compiler flags. We need to add it ourselves.

**Comment by romainguy** [comment]
Fixed by https://github.com/google/filament/commit/01f18caf53be2bac427bc12e733ccb35acbc2132

**Comment by rawnsley** [comment]
@romainguy I think [this test in the Android LogStream](https://github.com/google/filament/blob/main/libs/utils/src/Log.cpp#L50) needs to be modified as well.

**Comment by romainguy** [comment]
Thank! I fixed it.
